File: 344-reverse-string/reverse-string.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        def reverse(l, r):
            if l >= r:
                return
            s[r], s[l] = s[l], s[r]
            
            return reverse(l + 1, r - 1)

        reverse(0 , len(s) - 1) 

File: 344-reverse-string/reverse-string.go
func reverseString(s []byte)  {
    n := len(s)
    for i := 0; i < n / 2; i++{
        temp := s[i]
        s[i] = s[n - i - 1]
        s[n - i - 1] = temp
    }
    
}